src/engine/Match_tainting_mode.ml

```diff
@@ -914,8 +914,8 @@ let check_fundef lang options taint_config opt_ent ctx ?glob_env
   in
   (flow, mapping)
 
-let check_rule ?dep_matches per_file_formula_cache (rule : R.taint_rule)
-    match_hook (xconf : Match_env.xconfig) (xtarget : Xtarget.t) =
+let check_rule per_file_formula_cache (rule : R.taint_rule) match_hook
```

Comment:
We should not call it match_hook but matches_hook then.
I like that we moved SSC code out of here, but I don't like we piggypack on the
match_hook. Maybe it should be a separate hook, or maybe we should be able to do that 
in the caller.

Reply:
The original intent of match_hook was just give control to code up in the call stack to do something with a match, such as printing it the match, incrementally, without having to wait for the whole result.